cobc/warning.def

```diff
@@ -114,5 +114,8 @@ CB_ONWARNDEF (cb_warn_dialect, "dialect",
 CB_NOWARNDEF (cb_warn_source_after_code, "dangling-text",
 	_("  -Wdangling-text       warn about source text after program-area"))
 
+CB_ONWARNDEF (cb_missing_newline, "missing-newline",
```

Comment:
You may want to move this out as a separate minimal patch at https://sourceforge.net/p/gnucobol/patches/ - it cannot be "significant" enough for copyright but when you care enough about that to do the change+changelog+testsuite entry I'd include that soon.
Note: as [GCC dropped that warning altogether](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=14331) and the COBOL standard never required it (use of some preprocessors and editors did) let's only include it with `-Wall` (CB_WARNDEF) or `-Wextra` (CB_NOWARNDEF) and with the llvm name [`-Wnewline-eof`](https://clang.llvm.org/docs/DiagnosticsReference.html#wnewline-eof).